Kernel/TTY/SlavePTY.cpp

```diff
@@ -28,8 +28,10 @@ bool SlavePTY::unref() const
         m_list_node.remove();
         return true;
     });
-    if (did_hit_zero)
+    if (did_hit_zero) {
+        const_cast<SlavePTY&>(*this).before_removing();
```

Comment:
Hmm. I'm seeing some patterns here. This looks pretty much identical to the implementation of `ListedRefCounted::unref`. 

The only reason it can't be used is SlavePty already inherits from RefCountedBase, so we'd have some fun multi-inheritance nightmares if we tried to use it. not sure if virtual inheritance from RefCountedBase gives me any warmer fuzzies than this duplicated code :|